ignite-worker/src/main/resources/OSGI-INF/blueprint/horizon-ignite.xml

```diff
@@ -33,4 +33,30 @@
     </service-properties>
   </service>
 
+<!--  ========= Detector whiteboard and plugins ==========-->
+
+  <bean id="detectorRegistry" class="org.opennms.poc.ignite.worker.ignite.detectors.DetectorRegistryImpl">
+    <argument ref="blueprintBundleContext"/>
+    <argument ref="ignite"/>
+  </bean>
+
+<!--  TODO: not sure this is the right way to initialize a static bean. Feels.... odd-->
+  <bean id="staticDetectorRegistry" class="org.opennms.poc.ignite.worker.ignite.detectors.StaticDetectorRegistry">
```

Comment:
Constructor injection will do the trick, its fairly common (anti) pattern when lifecycle is out of control. I recall that other way was using `factory-method` which worked for Spring but been bridged also by blueprint. Haven't used it in a while, so its just an suggestion.